qt/scientific_interfaces/MultiDatasetFit/MDFAddWorkspaceDialog.cpp

```diff
@@ -89,6 +72,59 @@ void AddWorkspaceDialog::selectAllSpectra(int state) {
   }
 }
 
+QStringList AddWorkspaceDialog::availableWorkspaces() const {
+  auto &ADS = Mantid::API::AnalysisDataService::Instance();
+  QStringList workspaceNames;
+  auto wsNames = ADS.getObjectNames(Mantid::Kernel::DataServiceSort::Sorted);
+  for (auto name = wsNames.begin(); name != wsNames.end(); ++name) {
+    if (ADS.retrieveWS<Mantid::API::MatrixWorkspace>(*name)) {
+      workspaceNames << QString::fromStdString(*name);
+      continue;
+    }
+    auto grp = ADS.retrieveWS<Mantid::API::WorkspaceGroup>(*name);
+    if (grp) {
+      bool hasMatrixWorkspace = false;
+      for (auto ws : grp->getAllItems()) {
+        if (dynamic_cast<Mantid::API::MatrixWorkspace *>(ws.get())) {
+          hasMatrixWorkspace = true;
+          break;
+        }
+      }
+      if (hasMatrixWorkspace) {
+        workspaceNames << QString::fromStdString(*name);
+      }
+    }
+  }
+  workspaceNames.sort();
```

Comment:
Excuse my ignorance, but are the names not sorted by the call to `ADS.getObjectNames(Mantid::Kernel::DataServiceSort::Sorted)` already?